### PR TITLE
Tutorial update

### PR DESCRIPTION
**Pull request by stuartmcalpine**
Update the tutorials to give a smoother introduction to executions and pipelines (dependencies).

### CODE REVIEW

**Comment by stuartmcalpine** [comment]
When updating the schema table descriptions, what are:
- `is_external_link` and `is_valid` in the database table
- `supersede_date` in the alias table

**Review by JoanneBogart — CHANGES_REQUESTED** [review]
I've made a bunch of comments. In a few places I think something like the suggested rewording needs to be used to get the point across clearly, but overall it looks good and represents an impressive amount of work.

**Comment by stuartmcalpine** [comment]
Have updated with your comments and to include information on how to modify the default execution metadata when registering a dataset.

File: docs/source/tutorial_notebooks/getting_started.ipynb [hunk 1]
```diff
@@ -166,25 +169,30 @@
...
     "\n",
     "Datasets are registered under a given `owner`. This can be any string, however it should be informative. If no `owner` is specified, and no global `owner` was set when the `DataRegistry` instance was created, `$USER` is used as the default.\n",
     "\n",
-    "One further level of classification is the `owner_type`, which can be one of `user`, `group`, `project` or `production`. If `owner_type` is not specified, and no global `owner_type` was set when the `DataRegistry` instance was created, `user` is the default. Note that production datasets can never be overwritten.\n",
+    "One further level of classification is the `owner_type`, which can be one of `user`, `group`, `project` or `production`. If `owner_type` is not specified, and no global `owner_type` was set when the `DataRegistry` instance was created, `user` is the default.\n",
+    "\n",
+    "Note that `owner_type=production` datasets can only got into the production schema, and can never be overwritten (see the \"[Production Schema](https://github.com/LSSTDESC/dataregistry/tree/main/docs/source/tutorial_notebooks/production_scheme.ipynb)\" tutorial for more information on the production schema).\n",
```
Comment (JoanneBogart) [inline docs/source/tutorial_notebooks/getting_started.ipynb:185]: ...can only **go** into the production schema..

File: docs/source/tutorial_notebooks/getting_started.ipynb [hunk 2]
```diff
@@ -139,12 +142,12 @@
    },
    "outputs": [],
    "source": [
-    "# Create a empty text file\n",
+    "# Create a empty text file as some example data\n",
```
Comment (JoanneBogart) [inline docs/source/tutorial_notebooks/getting_started.ipynb:145]: Create **an** empty text file..

File: docs/source/tutorial_notebooks/getting_started_2.ipynb [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f48aec2e-2b35-49ed-be76-8818d9e79b2c",
+   "metadata": {},
+   "source": [
+    "## Executions\n",
+    "\n",
+    "The `dataregistry` nomenclature for a grouping of datasets is an \"execution\". For example, when multiple datasets are produced from a DESC pipeline stage, or numerical simulation, say, the parent pipeline stage or numerical simulation would be the \"execution\", and we associate the individual datasets as child members of that execution.\n",
```
Comment (JoanneBogart) [inline docs/source/tutorial_notebooks/getting_started_2.ipynb:50]: I don't think this first sentence gives the right flavor.   The phrase "a grouping of datasets" is very broad.  There are various ways to group datasets and various reasons for doing so. Some of them have to do with an execution (as I think of it) and others don't.   I think of an execution  as an abstraction of a particular run of a program (or script, pipeline stage, etc.. Choose your favorite term).   It associates inputs (one group of datasets) with outputs (another group of datasets) and has some characteristics of its own (e.g., the time the run started, configuration file if any, etc.).

File: docs/source/tutorial_notebooks/getting_started_2.ipynb [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "2d723a37-4101-496c-b385-0a2644aa7ad8",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Dependencies\n",
+    "\n",
+    "Executions represent groupings of datasets connected to a single \"run\" (such as the output from a single stage of a DESC pipeline). Datasets can also be linked to one another via \"dependencies\", to classify them as a precursor dataset within the scope of a larger pipeline. Dependencies are created between datasets through their execution, however note that not all the datasets within the precursor execution are required to be dependencies of the datasets in the following execution.\n",
```
Comment (JoanneBogart) [inline docs/source/tutorial_notebooks/getting_started_2.ipynb:171]: Prefer to substitute "define" for "represent".

**Review by JoanneBogart — CHANGES_REQUESTED** [review]
The main thing to address is the concept of "execution".

File: docs/source/tutorial_notebooks/getting_started_2.ipynb (hunk 3 above)
Comment (stuartmcalpine) [reply docs/source/tutorial_notebooks/getting_started_2.ipynb:50]: I've had another go at the wording

**Review by JoanneBogart — APPROVED** [review]
LGTM

**Pull request merged**